### DISCUSSION TITLE
How to run asynchronous tasks?

### DISCUSSION THREAD

**Discussion by hirale** [OpenMage/magento-lts#3723] [Questions & Answers] [answered]
We want to push some data from the backend to a third-party analysis service during the checkout process to ensure data integrity. There is always some transaction data missing on the client side, which may be caused by anti-tracking plug-ins or network problems.

Is there a recommended way to implement asynchronous tasks in OpenMage?

**Comment by luigifab** [comment, marked as answer]
There are several possibilities, depending on the amount of data to be synchronized.

- you can sync them in real time but after response sent to the client (#1592), perform sync from the `__destruct` of your model
- you can create a waiting list (a simple table with a list of customer_id/order_id) and a cron to perform sync (here it's easy to restart sync)

I used both methods for different purposes.
